include/pagmo/problems/cec2009.hpp

```diff
@@ -0,0 +1,873 @@
+/* Copyright 2017 PaGMO development team
+
+This file is part of the PaGMO library.
+
+The PaGMO library is free software; you can redistribute it and/or modify
+it under the terms of either:
+
+  * the GNU Lesser General Public License as published by the Free
+    Software Foundation; either version 3 of the License, or (at your
+    option) any later version.
+
+or
+
+  * the GNU General Public License as published by the Free Software
+    Foundation; either version 3 of the License, or (at your option) any
+    later version.
+
+or both in parallel, as here.
+
+The PaGMO library is distributed in the hope that it will be useful, but
+WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+for more details.
+
+You should have received codetail::pi()es of the GNU General Public License and the
+GNU Lesser General Public License along with the PaGMO library.  If not,
+see https://www.gnu.org/licenses/. */
+
+#ifndef PAGMO_PROBLEM_CEC2009_HPP
+#define PAGMO_PROBLEM_CEC2009_HPP
+
+#include <cassert>
+#include <stdexcept>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "../detail/constants.hpp"
+#include "../exceptions.hpp"
+#include "../problem.hpp" // needed for cereal registration macro
+#include "../types.hpp"
+
+#define MYSIGN(x) ((x) > 0 ? 1.0 : -1.0)
```

Comment:
Could we make this just a private static function inside the class? Or in the detail namespace at least (in class would be better if possible)?